tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -24,15 +27,50 @@ internal class Security : IDatadogSecurity, IDisposable
     {
         private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<Security>();
 
+        private static readonly Dictionary<string, string> RequestHeaders;
+        private static readonly Dictionary<string, string> ResponseHeaders;
+
         private static Security _instance;
         private static bool _globalInstanceInitialized;
         private static object _globalInstanceLock = new();
 
         private readonly IWaf _waf;
-        private readonly IAppSecAgentWriter _agentWriter;
         private readonly InstrumentationGateway _instrumentationGateway;
         private readonly SecuritySettings _settings;
 
+        static Security()
+        {
+            RequestHeaders = new Dictionary<string, string>()
```

Comment:
A `Dictionary` seems like overkill is you're never doing any lookups. A `List<T>` or `T[]` (where `T` is `KeyValuePair<string, string>`) would be a little more efficient in terms of CPU (initialization only?) and memory usage (forever static).

Whichever collection you end up with, it would also benefit a little from initializing with capacity since it's known at compile time:

```suggestion
            RequestHeaders = new Dictionary<string, string>(capacity: 18)
```

(same comment for `ResponseHeaders` below)